pyoozie/tags.py

```diff
@@ -494,6 +496,133 @@ def _xml(self, doc, tag, text):
         return doc
 
 
+class _WorkflowEntity(typing.Iterable):
+    """An abstract object representing an Oozie workflow action that can be serialized to XML."""
+    # pylint: disable=abstract-method
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(
+            self,
+            xml_tag,       # typing.Text
+            name=None,     # type: typing.Optional[typing.Text]
+            on_error=None  # type: typing.Optional[_WorkflowEntity]
+    ):
+        # type: (...) -> None
+        self.xml_tag = xml_tag
+        self.__name = name
+        self.__on_error = copy.deepcopy(on_error)
+        self.__uid = uuid.uuid4().hex[:8]
+        self.__identifier = self.create_identifier(xml_tag)
+
+    def create_identifier(self, xml_tag):
```

Comment:
This function should be private, unless you have a use case in mind for users creating identifiers to be used somewhere.